crates/sui-framework/packages/move-stdlib/tests/integer_tests.move

```diff
@@ -104,6 +106,16 @@ public(package) macro fun test_divide_and_round_up($max: _, $cases: vector<_>) {
     })
 }
 
+public(package) macro fun test_mul_div($max: _, $cases: vector<_>) {
+    let max = $max;
+    let cases = $cases;
+    assert_eq!(max.mul_div(max, max.max(1)), max);
+    cases!(max, cases, |_case_pred, case, _case_succ| {
+        assert_eq!(case.mul_div(case, case.max(1)), case); // avoid division by 0
+        // TODO: more test cases?
```

Comment:
Why not use `_case_pred` and `_case_succ`?

Reply:
Sorry, I don't think I was clear. 
The purpose of `case_pred` and `case_succ` is to build up invariants about predecessors and successors of `case`. 
Ideally we could kill those cases if we had `saturating_` for add/sub. 
But on `// TODO: more test cases?` these are the `_pred` and `_succ` cases are tools for testing these sorts of mathematical invariants you would expect for the operation being tested.